code/modules/mob/dead/observer/orbit.dm

```diff
@@ -65,7 +65,7 @@
 				var/obj/item/card/id/identification_card = M.get_idcard()
 				if (identification_card)
 					serialized["role_icon"] = "hud[ckey(identification_card.GetJobIcon())]"
-				else
+				else if(mind.assigned_role in SSjob.name_occupations)
```

Comment:
```suggestion
				else if(SSjob.name_occupations[mind.assigned_role])
```

Reply:
They do the same thing

Reply:
> They do the same thing

https://www.byond.com/docs/ref/#/operator/in

It does the same thing yeah, but it's related with time coplexity. `A in B` is actually the same as `for(var/each in B` and it is O(n) which means can possibly worse if the target job is at the end of the list, while `B[A]` directly refers the job from the list which is O(1)

Reply:
That's dumb